plugins/modules/site_workflow_manager.py

```diff
@@ -760,14 +992,266 @@ def get_want(self, config):
+
+        Returns:
+            dict: The API response from the 'updates_a_floor' operation.
+        """
+
+        try:
+            self.log("Updating floor with parameters: {0}".format(site_params), "INFO")
+            parent_name = site_params.get("site", {}).get("floor", {}).get("parentName")
+            type = site_params.get("type")
+            (parent_id) = self.get_parent_id(parent_name)
+            site_params['site']['floor']['parentId'] = parent_id
+            site_params['site']['floor']['unitsOfMeasure'] = "feet"
+            self.log("Updated site_params with parent_id: {0}".format(site_params), "INFO")
+            floor_param = site_params.get('site', {}).get('floor')
+            site_id = self.have.get("site_id")
+            floor_param['id'] = site_id
+            self.log(floor_param)
+
+            response = self.dnac._exec(
+                family="site_design",
+                function='updates_a_floor',
+                op_modifies=True,
+                params=floor_param
+            )
+            self.log(response)
+            if response and isinstance(response, dict):
+                self.log("Initial API response from 'updates_a_floor': {0}".format(response), "DEBUG")
+                executionid = response.get("executionId")
+
+                if executionid:
+                    while True:
+                        execution_details = self.get_execution_details(executionid)
+                        if execution_details.get("status") == "SUCCESS":
+                            self.log("Floor update completed successfully.", "INFO")
+                            break
+                        elif execution_details.get("bapiError"):
+                            self.log("Error during floor update execution: {0}".format(execution_details.get("bapiError")), "ERROR")
+                            self.module.fail_json(msg=execution_details.get("bapiError"), response=execution_details)
+                            break
+
+            return response
+        except Exception as e:
+            error_msg = "Exception occurred while updating floor with site_id '{0}' due to: {1}".format(
+                site_params.get('site_id'), str(e)
+            )
+            self.log(error_msg, "ERROR")
+        return self
+
+    def update_building(self, site_params):
+        """
+        Updates a building in the site hierarchy using the provided site id.
+
+        Args:
+            site_params (dict): Dictionary containing parameters required for the building update, including the site_id.
+
+        Returns:
+            dict: The API response from the 'update_a_building' operation.
+        """
+        try:
+            self.log("Updating building with parameters: {0}".format(site_params), "INFO")
+            parent_name = site_params.get("site", {}).get("building", {}).get("parentName")
+            type = site_params.get("type")
+            (parent_id) = self.get_parent_id(parent_name)
```

Comment:
remove () around the variable - parent_id.

Reply:
Removed

Reply:
still it is enclosed with braces...
 (parent_id) = self.get_parent_id(parent_name)